pennylane/math/multi_dispatch.py

```diff
@@ -846,12 +846,46 @@ def norm(tensor, like=None, **kwargs):
             axis_val = kwargs.pop("axis")
             kwargs["dim"] = axis_val
 
+    elif like == "autograd":
+        ord = kwargs.get("ord", None)
+        if ord is not None and ord != "fro" and ord != "nuc":
+            norm = _autograd_norm
+        else:
+            from autograd.numpy.linalg import norm
+
     else:
         from scipy.linalg import norm
 
     return norm(tensor, **kwargs)
 
 
+def _autograd_norm(tensor, **kwargs):
+    """Autograd differentiable norm"""
+    if kwargs.get("axis", None) is not None:
+        raise ValueError("Specifying axes is not currently supported.")
+    if kwargs.get("keep_dims", False):
+        raise ValueError("Setting keep_dims is not currently supported.")
+
+    import autograd.numpy as anp
+
+    ord = kwargs.get("ord", 2)
+
+    if ord == 0:
+        return anp.sum(tensor != 0)
+
+    if ord == onp.inf:
+        _norm = 0.0
+        for val in tensor.flatten():
+            aval = anp.abs(val)
+            _norm = aval if aval > _norm else _norm
+        return _norm
+
+    _norm = 0.0

```

Comment:
Should we check/assert that ord is now 2?

Reply:
Or is this for any `p`-norm?

Reply:
It's for any `p`-norm, but it's irrelevant now.